lnst/Recipes/ENRT/VlansOverBondRecipe.py

```diff
@@ -125,12 +129,8 @@ def test_wide_configuration(self):
         host2.vlan1 = VlanDevice(realdev=host2.eth0, vlan_id=self.params.vlan1_id)
         host2.vlan2 = VlanDevice(realdev=host2.eth0, vlan_id=self.params.vlan2_id)
 
-        configuration = super().test_wide_configuration()
-        configuration.test_wide_devices = []
-        for host in [host1, host2]:
-            configuration.test_wide_devices.extend([host.vlan0,
-                host.vlan1, host.vlan2])
-        configuration.test_wide_devices.append(host1.bond0)
+        config = super().test_wide_configuration()
+        config.track_device(host1.bond0)
```

Comment:
So, my understanding of track/untrack_device methods is to simply get a device into `configured_devices` so that it can be processed in `generate_test_wide_description()` if needed. However, in particular in this recipe, the description method directly references the device where required. So, it's not needed to track it here.

Similarly in other recipes you can simply do self.configured_devices + [self.matched.host1.some_device].

I'm not quite sure if this method is useful at all. Some explanation would be good here.

Reply:
Ondrej explained the concept to me offline. However, I have one rename proposal. Can't we unify the `track/untrack_device()` and `configure_and_register()` methods to either:
* `register/unregister_device()` and `configure_and_register()` or
* `track/untrack_device()` and `configure_and_track()`

Simply to know that these methods are related.

@olichtne wdyt?

Reply:
> my understanding of track/untrack_device methods is to simply get a device into `configured_devices` so that it can be processed in generate_test_wide_description() if needed.

Your understanding is correct. Some recipes added some devices to `self.test_wide_devices` even though the devices in question didn't get any IP assigned. I mirrored this behaviour to the `track_device()` method. However, I think this could be eventually unified. Ondrej shared his thoughts that he'd like to keep a track of every configured device in there, but I'm not sure how we would do that yet.

I'd definitely leave this out of this MR though, so I'll just rename the ip method to `configure_and_track_ip()` for now. Later I think, most recipes should be cleaned up. There's a lot of unnecessary duplication.

Reply:
ack to `configure_and_track_ip()`

Reply:
They have been renamed and this can be resolved.